doc/release/upcoming_changes/19550.new_feature.rst

```diff
@@ -0,0 +1,12 @@
+Use f-strings to choose printing options for numeric dtypes
+-----------------------------------------------------------
+
+You can use f-strings now to format the precision, sign and the use of exponential notation for numeric arrays with a syntax inspired from [Python Format Specification Mini-Language](https://docs.python.org/3/library/string.html#formatspec).
+
+    >>> a = np.array([1.001, -0.101, np.pi, -np.e])
+    >>> print(f"{a:.2f}")
+    [ 1.   -0.1   3.14 -2.72]
+    >>> print(f"{a:+.2e}")
+    [+1.00e+00 -1.01e-01 +3.14e+00 -2.72e+00]
+
+See ``_parse_format_spec`` and ``array2string`` for reference.
```

Comment:
Let's make this a link and remove the private function
```suggestion
See `numpy.array2string` for reference.
```

Reply:
I actually think is useful to see the documentation for the private function to be clear which parameter of the f string maps to which parameter in array2string.

Do you think of a better way to achieve that?

Reply:
I suggested elsewhere that we add a [how to](https://numpy.org/devdocs/user/howtos_index.html) on printing arrays since there are so many ways to do it. This would be a good place to explain the minilanguage. We cannot link to the docstring of a private function.